source/src/localisation/cs_TrackableParser.cpp

```diff
@@ -124,136 +223,371 @@ void TrackableParser::handleScannedDevice(scanned_device_t* device) {
+	return ERR_SUCCESS;
 }
 
-cs_ret_code_t TrackableParser::handleGetFilterSummariesCommand(
-		trackable_parser_cmd_get_filer_summaries_t* cmd_data) {
-	// TODO(Arend): implement later.
-	return ERR_NOT_IMPLEMENTED;
+void TrackableParser::handleGetFilterSummariesCommand(
+		trackable_parser_cmd_get_filter_summaries_t* cmd_data, cs_result_t& result) {
+	LOGTrackableParserDebug("handle get filter summaries:");
+	for (auto& trackingFilter : _parsingFilters) {
+		if (trackingFilter == nullptr) {
+			LOGTrackableParserDebug("filter: nullptr");
+			continue;
+		}
+		LOGTrackableParserDebug("filter: %u", trackingFilter->runtimedata.filterId);
+	}
+
+	// stack allocate a buffer summaries object fitting at most max summaries:
+	auto requiredBuffSize = sizeof(trackable_parser_cmd_get_filter_summaries_ret_t)
+							+ sizeof(tracking_filter_summary_t) * _parsingFiltersCount;
+
+	if (result.buf.len < requiredBuffSize) {
+		result.returnCode = ERR_BUFFER_TOO_SMALL;
+		return;
+	}
+
+	// placement new constructs the object in the buff and now has enough space for the summaries.
+	auto retvalptr  = new (result.buf.data) trackable_parser_cmd_get_filter_summaries_ret_t;
+	result.dataSize = requiredBuffSize;
+
+	retvalptr->masterCrc     = _masterHash;
+	retvalptr->masterVersion = _masterVersion;
+	retvalptr->freeSpace     = getTotalHeapAllocatedSize();
+
+	for (size_t i = 0; i < _parsingFiltersCount; i++) {
+		retvalptr->summaries[i].id      = _parsingFilters[i]->runtimedata.filterId;
+		retvalptr->summaries[i].crc     = _parsingFilters[i]->runtimedata.crc;
+		retvalptr->summaries[i].flags   = _parsingFilters[i]->metadata.flags;
+		retvalptr->summaries[i].version = _parsingFilters[i]->metadata.version;
+	}
+
+	result.returnCode = ERR_SUCCESS;
+}
+
+// -------------------------------------------------------------
+// ---------------------- Utility functions --------------------
+// -------------------------------------------------------------
+void TrackableParser::startProgress() {
+	filterModificationInProgress = true;
+	_masterVersion               = 0;
+}
+
+uint16_t TrackableParser::masterCrc() {
+	uint16_t masterCrc = 0;
+	uint16_t* prevCrc  = nullptr;
+
+	for (auto trackingFilter : _parsingFilters) {
+		if (trackingFilter == nullptr) {
+			break;
+		}
+
+		LOGTrackableParserDebug("filter crc: %x", trackingFilter->runtimedata.crc);
+
+		// appends/applies this filters' crc onto master crc
+		masterCrc =
+				crc16(reinterpret_cast<const uint8_t*>(&trackingFilter->runtimedata.crc),
+					  sizeof(trackingFilter->runtimedata.crc),
+					  prevCrc);
+
+		prevCrc = &masterCrc;  // first iteration will have prevCrc == nullptr, all subsequent ones point to masterCrc.
+	}
+
+	LOGTrackableParserDebug("master crc: %x", masterCrc);
+
+	return masterCrc;
+}
+
+bool TrackableParser::checkFilterSizeConsistency() {
+	bool consistencyChecksFailed = false;
+
+	size_t index = 0;
+	while (index < _parsingFiltersCount) {
```

Comment:
Why not a for loop? You can also skip the i++ in a for loop.

Reply:
It would be very tempting to move the `index++` at the end of the loop into the empty slot of a for loop. That would break the logic, so I added an extra hurdle.